code/modules/mob/living/carbon/human/appearance.dm

```diff
@@ -242,3 +317,47 @@
 			valid_facial_hairstyles += facialhairstyle
 
 	return valid_facial_hairstyles
+
+/mob/living/carbon/human/proc/generate_valid_head_accessories()
+	var/list/valid_head_accessories = new()
+	var/obj/item/organ/external/head/H = get_organ("head")
+	for(var/head_accessory in head_accessory_styles_list)
+		var/datum/sprite_accessory/S = head_accessory_styles_list[head_accessory]
+
+		if(!(H.species.name in S.species_allowed)) //If the user's head is not of a species the head accessory style allows, skip it. Otherwise, add it to the list.
+			continue
+		valid_head_accessories += head_accessory
+
+	return valid_head_accessories
+
+/mob/living/carbon/human/proc/generate_valid_markings()
+	var/list/valid_markings = new()
+	var/obj/item/organ/external/head/H = get_organ("head")
+	for(var/marking in marking_styles_list)
+		var/datum/sprite_accessory/S = marking_styles_list[marking]
+
+		if(!(species.name in S.species_allowed)) //If the user's head is not of a species the marking style allows, skip it. Otherwise, add it to the list.
+			continue
+		if(H.species.flags & ALL_RPARTS) //If the user is a species that can have a robotic head...
+			var/datum/robolimb/robohead = all_robolimbs[H.model]
+			if(!(S.models_allowed && (robohead.company in S.models_allowed))) //Make sure they don't get markings incompatible with their head.
+				continue
+		valid_markings += marking
+
+	return valid_markings
+
+/mob/living/carbon/human/proc/generate_valid_body_accessories()
+	var/list/valid_body_accessories = new()
+	for(var/B in body_accessory_by_name)
+		var/datum/body_accessory/A = body_accessory_by_name[B]
+		if(check_rights(R_ADMIN, 1, src))
```

Comment:
Means admins playing characters that are of species with tails will be able to choose the snake body accessory with CMA and the Nano-mirrors.